src/main/util.cpp

```diff
@@ -55,9 +55,89 @@ bool segvSpin = false;
 size_t cvc4StackSize;
 void* cvc4StackBase;
 
+/* Computes the length of a string. Safe to use in a signal handler. */
+size_t safe_strlen(const char* str) {
+  size_t l = 0;
+  while (*str) {
+    str++;
+    l++;
+  }
+  return l;
+}
+
+/* Writes a string to fd. Safe to use in a signal handler. */
+void safe_print(int fd, const char* msg) {
+  write(fd, msg, safe_strlen(msg));
```

Comment:
Makes sense. My suggestion would be to add a `safe_strnlen` and a `safe_nprint` function that each take a length as an argument. The idea being that `safe_print` should only be used for string literals (because getting the length of a string literal is tricky/annoying at compile time) and `safe_nprint` for printing buffers. Does that make sense?